tests/main/cgroup-freezer/task.yaml

```diff
@@ -7,19 +7,26 @@ prepare: |
     install_local test-snapd-sh
 execute: |
     # Start a "sleep" process in the background
-    test-snapd-sh -c 'exec sleep 1h' &
+    test-snapd-sh -c 'touch $SNAP_DATA/1.stamp && exec sleep 1h' &
     pid1=$!
-    # Give snap-confine a moment to start and perform its task.
-    sleep 3
+    # Ensure that snap-confine has finished its task and that the snap process
+    # is active. Note that we don't want to wait forever either.
+    for i in $(seq 30); do
```

Comment:
(nitpick^2) You could make this a sh function and reuse it in both places. But fine, thanks for this update.

Reply:
I think it's fine as is but I'll consider it for the next pass :)